### PR TITLE
Agent ID cards now jump to the front of wallets and can disguise other more powerful ID cards.

### PR DESCRIPTION
**Pull request by Timberpoes**
## About The Pull Request

Many moons after I reworked ID cards, I feel the Agent ID card needs a little bit of extra love.

This change adds an option during forging which allows the user to specify whether the agent ID card will force itself to the front-facing/visible ID slot in wallets.

This provides traitors with a way to effectively disguse any special ID cards and accesses they may have without making the agent ID card a must-buy TC tax. If your traitor strategy involves gathering many ID cards, this is a definite buff to an item that took a heavy hit to power in the rework.

## Why It's Good For The Game

I feel the agent ID card is a bit underpowered and could use a boost in utility. This should give it that boost without defeating the whole point of the ID card and access overhaul.

## Changelog
:cl:
balance: The traitor's Agent ID card has a new option available during forging called Wallet ID Spoofing. Selecting this option will automatically snap the Agent ID card to the front of any wallet, allowing the traitor to hide more valuable ID cards behind it. This is disabled when the card is unforged.
/:cl:

### CODE REVIEW

**Comment by OmegaDarkPotato** [comment]
I know that bringing anything policy-related into a discussion on code'll only serve to leave me looking like an idiot, but because we're allowed to know how antagonists work even on MRP, won't this lead to sec simply searching anyone with a wallet for free valids? It's not like people use wallets nowadays anyway for purposes other than ghetto AA with 4 silver head cards.

**Comment by brotherbeyondd** [comment]
> I know that bringing anything policy-related into a discussion on code'll only serve to leave me looking like an idiot, but because we're allowed to know how antagonists work even on MRP, won't this lead to sec simply searching anyone with a wallet for free valids? It's not like people use wallets nowadays anyway for purposes other than ghetto AA with 4 silver head cards.
> 
> EDIT: Not that I'm arguing AGAINST this change, the cham card definitely needs some sort of tangible buff, I'm just kinda waiting for that promise of it in a chameleon PDA too.

its the same as seccies checking people's ids regularly. I believe agent ids can still scan other ids so having a wallet provides literally fuckall since you dont need the other ids so :shrug:

**Comment by OmegaDarkPotato** [comment]
> its the same as seccies checking people's ids regularly. I believe agent ids can still scan other ids so having a wallet provides literally fuckall since you dont need the other ids so 🤷

worked with agent ids recently? they have like, 5 slots, max.
they can't do crap aside from open maybe 6 doors on the station, the other IDs are absolutely essential.

**Comment by Timberpoes** [comment]
> 
> 
> I know that bringing anything policy-related into a discussion on code'll only serve to leave me looking like an idiot, but because we're allowed to know how antagonists work even on MRP, won't this lead to sec simply searching anyone with a wallet for free valids? It's not like people use wallets nowadays anyway for purposes other than ghetto AA with 4 silver head cards.
> 
> EDIT: Not that I'm arguing AGAINST this change, the cham card definitely needs some sort of tangible buff, I'm just kinda waiting for that promise of it in a chameleon PDA too.

FNR searching of players is already against the rules.

Carrying a wallet is not enough to conduct any search without some other really good justification as to why you need to conduct any search at all. It costs a few leather to make from botany, some lockers have them at shift start and I'd argue more validhunters and powergamers carry them than antags.

File: code/game/objects/items/storage/wallets.dm [hunk 1]
```diff
@@ -64,6 +64,12 @@
 		var/obj/item/card/id/id_card = card
 		if(!istype(id_card))
 			continue
+
+		// Chameleon IDs explicitly jump to the front so they can disguise other cards in wallets.
+		if(istype(id_card, /obj/item/card/id/advanced/chameleon))
```
Comment (coiax) [inline code/game/objects/items/storage/wallets.dm:69]: I thought  `istype` checks were frowned upon

Comment (Timberpoes) [reply code/game/objects/items/storage/wallets.dm:69]: It's inevitable in this scenario. Wallets can hold items other than ID cards, so I'll need some sort of typed loop.

I can replement this with a typed for loop, or an `as anything` for loop as I'd still need istype checks.

**Comment by Technoturnovers** [comment]
I carry a wallet because it looks nice, and because I don't put my ID in my PDA if possible because of PDA bombs.

File: code/game/objects/items/storage/wallets.dm (hunk 1 above)
Comment (coiax) [reply code/game/objects/items/storage/wallets.dm:69]: I mean, did you consider marking the functionality of "this ID goes to the front of the wallet" in some way other than behaviour assigned to a type? A trait?

Comment (Timberpoes) [reply code/game/objects/items/storage/wallets.dm:69]: I didn't consider that. I could add it as a trait.

**Comment by Timberpoes** [comment]
This has been re-imagined as an item trait. Chameleon cards can have this trait added to them during forging, making the functionality optional. Best of both worlds.

**Pull request merged**